src/driver-sinowealth.c

```diff
@@ -71,7 +71,13 @@ enum rgb_effect {
 	RGB_BREATHING1 = 0xa, /* single color breathing */
 	RGB_TAIL = 0x4,
 	RGB_RAVE = 0x7,
-	RGB_WAVE = 0x9
+	RGB_WAVE = 0x9,
+
+	/* The value mice with no LEDs have.
+	* Unreliable as non-constant.
+	* Do **not** overwrite it.
+	 */
+	RGB_NOT_SUPPORTED = 0xff,
```

Comment:
If it's not constant, how about this approach:
```c
enum rgb_effect {
   ....
   RGB_WAVE = 0x9,
   _RGB_EFFECT_UNSUPPORTED /* Usually 0xff but it's unreliable so let' assume anything not know is unsupoprted */
};
```
and later

```c
if (config->rgb_effect < _RGB_EFFECT_UNSUPPORTED) {
```
This way anything we don't explicitly support becomes an unsupported one without a magic value

Reply:
I like current approach better, because if there happen to be lighting modes we don't know about (and there are such, currently), the user would not be able to change lighting modes at all.

Reply:
iirc in other drivers we just transparently map the unknown mode to a known one and pretend everything is ok. As long as you don't touch it it (should) stay as-is even if it's technically listed as wrong mode.

But yeah, we do really need a `RATBAG_LED_UNKNOWN`  enum value for this case...